src/replicache-on-after-commit.test.ts

```diff
@@ -0,0 +1,76 @@
+import {initReplicacheTesting, replicacheForTesting} from './test-util';
+import type {WriteTransaction} from './mod';
+import type {JSONValue} from './json';
+import {expect} from '@esm-bundle/chai';
+import * as sinon from 'sinon';
+
+initReplicacheTesting();
+
+async function addData(tx: WriteTransaction, data: {[key: string]: JSONValue}) {
+  for (const [key, value] of Object.entries(data)) {
+    await tx.put(key, value);
+  }
+}
+
+test('onAfterCommit', async () => {
+  const rep = await replicacheForTesting('on-after-commit', {
+    mutators: {addData, del: (tx, key) => tx.del(key)},
+  });
+
+  const spy = sinon.spy();
+  rep.onAfterCommit = spy;
+
+  await rep.mutate.addData({a: 1, b: 2});
+
+  expect(spy.callCount).to.equal(1);
+  expect(spy.lastCall.args).to.deep.equal([
+    [
+      {
+        op: 'add',
+        key: 'a',
+        newValue: 1,
+      },
+      {
+        op: 'add',
+        key: 'b',
+        newValue: 2,
+      },
+    ],
+    '9lrb08p9b7jqo8oad3aef60muj4td8ke',
+    't/000000000000000000000000000001',
+  ]);
+
+  spy.resetHistory();
+  await rep.mutate.addData({a: 1, b: 2});
+  expect(spy.callCount).to.equal(0);
+
+  await rep.mutate.addData({a: 11});
+  expect(spy.callCount).to.equal(1);
+  expect(spy.lastCall.args).to.deep.equal([
+    [
+      {
+        op: 'change',
+        key: 'a',
+        newValue: 11,
+        oldValue: 1,
+      },
+    ],
+    't/000000000000000000000000000003',
+    't/000000000000000000000000000005',
+  ]);
+
+  spy.resetHistory();
+  await rep.mutate.del('b');
+  expect(spy.callCount).to.equal(1);
+  expect(spy.lastCall.args).to.deep.equal([
+    [
+      {
+        op: 'del',
+        key: 'b',
+        oldValue: 2,
+      },
+    ],
+    't/000000000000000000000000000005',
+    't/000000000000000000000000000007',
```

Comment:
These temp hashes are a bit ugly.

One concern here is that once we get a persist, oldHash is not going to match newHash from the last onAfterCommit.

Reply:
Seems temp hashes break the whole purpose of exposing these hashes in this api.